src/provider/systemdmanager/service.rs

```diff
@@ -0,0 +1,321 @@
+        // an Error. If any error occurred, iteration stops on the first error and returns
+        // that in the outer result.
+        let env_variables = if let Some(vars) = container.env() {
+            debug!(
+                "Got environment vars: {:?} service {}",
+                vars, pod_state.service_name
+            );
+            let render_result = vars
+                .iter()
+                .map(|env_var| {
+                    // Replace variables in value
+                    CreatingConfig::render_config_template(
+                        &template_data,
+                        &env_var.value.as_deref().unwrap_or_default(),
+                    )
+                    .map(|value| (env_var.name.clone(), value))
+                })
+                .collect();
+
+            // If any single rendering failed, the overall result for the map will have
+            // collected the Err which we can check for here
+            match render_result {
+                Ok(rendered_values) => rendered_values,
+                Err(error) => {
+                    error!("Failed to render value for env var due to: {:?}", error);
+                    return Err(PodValidationError {
+                        msg: String::from("Failed to render a template"),
+                    });
+                }
+            }
+        } else {
+            // No environment variables present for this container -> empty vec
+            debug!(
+                "No environment vars set for service {}",
+                pod_state.service_name
+            );
+            vec![]
+        };
+        debug!(
+            "Setting environment for service {} to {:?}",
+            pod_state.service_name, &env_variables
+        );
+
+        Ok(env_variables)
+    }
+
+    fn get_command(container: &Container, pod_state: &PodState) -> Result<String, StackableError> {
+        // Return an error if no command was specified in the container
+        // TODO: We should discuss if there can be a valid scenario for this
+        // This clones because we perform some in place mutations on the elements
+        let mut command = match container.command() {
+            Some(command) => command.clone(),
+            _ => {
+                return Err(PodValidationError {
+                    msg: format!(
+                    "Error creating systemd unit for container {}, due to missing command element.",
+                    container.name()
+                ),
+                })
+            }
+        };
+
+        let package_root = pod_state.get_service_package_directory();
+
+        trace!(
+            "Commmand before replacing variables and adding packageroot: {:?}",
+            command
+        );
+        // Get a mutable reference to the first element of the command array as we might need to
+        // add the package directory to this to make it an absolute path
+        let binary = match command.get_mut(0) {
```

Comment:
This match could be changed to a .... `ok_or_else`? I believe...

Reply:
Unless I am mistaken that would not work with the early return?